src/finchlite/finch_assembly/nodes.py

```diff
@@ -101,6 +101,31 @@ def __repr__(self) -> str:
         return literal_repr(type(self).__name__, asdict(self))
 
 
+@dataclass(eq=True, frozen=True)
+class SSAVariable(AssemblyExpression):
+    """
+    Represents an SSA form of Variable assembly node that refers to variable
+    named `name`, with type `type`, and has ID `id`.
+
+    Attributes:
+        name: The name of the variable that it references.
+        type: The type of the variable that it references.
+        id: The ID of the variable
+    """
+
+    name: str
+    type: Any
```

Comment:
Instead of separating name and type here, can we just have a var node that holds the variable? This will make it easier to process the IR, as we don't need to duplicate cases for SSAVariable and Variable.